starlite/types/parsed_signature.py

```diff
@@ -75,7 +75,60 @@ def is_optional(self) -> bool:
     @property
     def is_collection(self) -> bool:
         """Whether the annotation is a collection type or not."""
-        return self.origin and issubclass(self.origin, Collection)
+        return bool(self.origin and self.origin is not Union and issubclass(self.origin, Collection))
+
+    def is_type_of(self, cl: type[Any]) -> bool:
+        """Whether the annotation is a subclass of the given type.
+
+        Args:
+            cl: The type to check.
+
+        Returns:
+            Whether the annotation is a subclass of the given type.
+        """
+        if self.origin:
+            if self.origin is Union:
+                return any(t.is_type_of(cl) for t in self.inner_types)
+            return issubclass(self.origin, cl)
+        return self.annotation is not Any and issubclass(self.annotation, cl)
+
+    def has_type_of(self, cl: type[Any]) -> bool:
```

Comment:
Maybe `has_any_type`? It doesn't fit in as good with the `is_<thing>_of` scheme, but it's a bit more explicit.

Reply:
Although, any (as in `Any`) is a bit of an overloaded word in the typing sphere. Let me sleep on it:)

Reply:
> Although, any (as in `Any`) is a bit of an overloaded word in the typing sphere

Oh, that's right. Hadn't considered that. I was thinking of the `any` builtin.

Reply:
`has_inner_type()` (we use "inner_type" for the `ParsedType` attribute)?

`contains_type()`?

Reply:
`contains_type` sounds good. Also makes it clear that this is intended to work on container types.